File: backend/src/main/java/kr/co/programmers/collabond/api/user/interfaces/UserMapper.java
package kr.co.programmers.collabond.api.user.interfaces;

import kr.co.programmers.collabond.api.user.domain.User;
import kr.co.programmers.collabond.api.user.domain.dto.SignUpResponseDto;
import kr.co.programmers.collabond.api.user.domain.dto.UserResponseDto;

public class UserMapper {

    public static UserResponseDto toResponseDto(User user) {
        return UserResponseDto.builder()
                .id(user.getId())
                .email(user.getEmail())
                .nickname(user.getNickname())
                .role(user.getRole().name())
                .createdAt(user.getCreatedAt())
                .updatedAt(user.getUpdatedAt())
                .build();
    }

    public static SignUpResponseDto toSignUpResponseDto(User user, String accessToken) {
        return SignUpResponseDto.builder()
                .nickname(user.getNickname())
                .role(user.getRole().name())
                .accessToken(accessToken)
                .build();
    }
}

File: backend/src/main/java/kr/co/programmers/collabond/api/user/interfaces/UserMapper.kt
package kr.co.programmers.collabond.api.user.interfaces

import kr.co.programmers.collabond.api.user.domain.SignUpResponseDto
import kr.co.programmers.collabond.api.user.domain.User
import kr.co.programmers.collabond.api.user.domain.UserResponseDto

fun toResponseDto(user: User) = UserResponseDto(
    id = user.id,
    email = user.email,
    nickname = user.nickname,
    role = user.role.name,
    createdAt = user.createdAt,
    updatedAt = user.updatedAt
)

fun toSignUpResponseDto(user: User, accessToken: String) = SignUpResponseDto(
    nickname = user.nickname,
    role = user.role.name,
    accessToken = accessToken
)